### DISCUSSION TITLE
Dark map timing

### DISCUSSION THREAD

**Discussion by boothym** [streetcomplete/StreetComplete#3591] [Q&A]
How does StreetComplete decide when to turn its map dark? Because I was out editing about 50 minutes after sunset recently and it still wasn't dark yet. 

This was even more annoying in the previous version the lighting quest would only appear when the map went dark, despite the street lights having already turned on.

**Comment by Helium314** [comment]
If I read the code correctly, the Android system decides whether to apply light or dark theme.
Do you have any other apps that switch between dark and light at a different time?

**Comment by matkoniecz** [comment]
SC has three options: always dark, always light, follow system instructions

Maybe system time for night mode is configurable on your phone?

I know that some apps decide on their own when to switch between light and dark, but SC is not doing this.

**Reply by Helium314** [reply]
There is also 'auto' mode, maybe not available on all devices.

**Reply by matkoniecz** [reply]
by "follow system instructions" I meant auto mode

**Reply by Helium314** [reply]
Well, there still is a fourth option:
https://github.com/streetcomplete/StreetComplete/blob/3f7a787ea61e5a565edffbf2b2b9f66b2945eb1e/app/src/main/java/de/westnordost/streetcomplete/Prefs.kt#L50-L55

To me it looks like actually the system *always* decides the theme, and SC just asks: Dear system, please set my theme to:
* `LIGHT` -> normal theme
* `DARK` -> night theme
* `AUTO` -> normal theme during day, dark theme at night
* `SYSTEM` -> whatever system is currently using

(and apparently `MODE_NIGHT_AUTO` and switching by time is deprecated, see https://developer.android.com/reference/androidx/appcompat/app/AppCompatDelegate#MODE_NIGHT_AUTO_TIME)

**Reply by matkoniecz** [reply]
that is what I get for not remembering, thinking that I remember and not checking code

**Reply by boothym** [reply]
I also use Twitter which has an automatic dark mode at sunset, it does the same thing as with StreetComplete.

I changed my phone's clock, and it took until 16:27 for Twitter and StreetComplete to change from light to dark, despite sunset being at 15:37. For sunrise it was similar, it switched about 45 minutes before.

My phone is a bit older and I can't see any dark mode settings, so perhaps it's not using sunrise/sunset but dawn/dusk (civil twilight) instead?

**Reply by Helium314** [reply]
No idea when Android is supposed to switch. On my phone the setting just says "Automatic (time of the day)"

**Reply by Lee-Carre** [reply]
I notice that this question is recent, and the experience I'm drawing on is old (over a year, maybe 2 years at this point). So, my recollections may no longer be relevant.

In the past, I used to have the preference set to `Auto` (because one of the following: it seemed sensible to an inexperienced newbie, the `System` option wasn't (yet) available (this was Android 9 & SC (circa) v25.1), Android didn't auto-toggle based on Sol (or I hadn't discovered that possibility, yet)). `Auto` toggled based on ambient light intensity (would use the dark theme when going through a tunnel during the day, but the light theme before & after). A problem with this, was that it would frequently switch with rapid large changes in ambient light.

However, for quite a while I've set it to `System` with the OS configured to toggle based on local sunrise & sunset, which achieves a good result for me.
